### PR TITLE
local_to_global, for set_bus and change_bus

### PR DESCRIPTION
**Pull request by Farid-Najar**

### CODE REVIEW

File: grid2op/tests/test_ma.py [hunk 1]
```diff
@@ -597,6 +599,273 @@ def check_mask_topo_vect(self, ma_env, domain, space="action"):
...
+        np.random.seed(0)
+        for agent in self.ma_env.agents:
+            # Test for loads
+            local_load_id = np.random.randint(0, self.ma_env._subgrids_cls['action'][agent].n_load)
+            local_act = self.ma_env.action_spaces[agent]({
+                'set_bus' : 
+                    (self.ma_env.action_spaces[agent].load_pos_topo_vect[local_load_id], 2)
+            })
+            #local_act = self.ma_env.action_spaces[agent]({})
+            #local_act.load_set_bus = [(local_load_id, 2)]
```
Comment (BDonnot) [inline grid2op/tests/test_ma.py:661]: use this instead !

File: grid2op/tests/test_ma.py [hunk 2]
```diff
@@ -597,6 +599,273 @@ def check_mask_topo_vect(self, ma_env, domain, space="action"):
...
+            #local_act.gen_set_bus = [(local_gen_id, 2)]
+            global_act = self.ma_env._local_action_to_global(agent, local_act)
+            global_gen_id = self.ma_env._subgrids_cls['action'][agent].gen_orig_ids[local_gen_id]
+            ref_global_act = self.ma_env._cent_env.action_space({
+                'set_bus' : 
+                    (self.ma_env._cent_env.action_space.gen_pos_topo_vect[global_gen_id], 2)
+            })
+            assert (global_act.set_bus == ref_global_act.set_bus).all()
+            assert global_act._modif_set_bus == ref_global_act._modif_set_bus
+            assert global_act == ref_global_act
```
Comment (BDonnot) [inline grid2op/tests/test_ma.py:688]: same: loop for all generators, use the "local_act.gen_set_bus" syntax

File: grid2op/tests/test_ma.py (hunk 1 above)
Comment (Farid-Najar) [reply grid2op/tests/test_ma.py:661]: Ok

File: grid2op/tests/test_ma.py [hunk 3]
```diff
@@ -597,6 +599,273 @@ def check_mask_topo_vect(self, ma_env, domain, space="action"):
...
+                ==\
+                    ma_env._subgrids_cls['action'][agent].interco_pos_topo_vect).all() 
+        
+    
+    def test_local_action_to_global_set_bus(self):
+        # TODO 
+        np.random.seed(0)
+        for agent in self.ma_env.agents:
+            # Test for loads
+            local_load_id = np.random.randint(0, self.ma_env._subgrids_cls['action'][agent].n_load)
```
[the remark this answers is not in the document]
Comment (Farid-Najar) [reply grid2op/tests/test_ma.py]: ok

File: grid2op/tests/test_ma.py [hunk 4]
```diff
@@ -597,6 +599,273 @@ def check_mask_topo_vect(self, ma_env, domain, space="action"):
...
+            #local_act.line_or_set_bus = [(local_line_or_id, 2)]
+            global_act = self.ma_env._local_action_to_global(agent, local_act)
+            global_line_or_id = self.ma_env._subgrids_cls['action'][agent].line_orig_ids[local_line_or_id]
+            ref_global_act = self.ma_env._cent_env.action_space({
+                'set_bus' : 
+                    (self.ma_env._cent_env.action_space.line_or_pos_topo_vect[global_line_or_id], 2)
+            })
+            assert (global_act.set_bus == ref_global_act.set_bus).all()
+            assert global_act._modif_set_bus == ref_global_act._modif_set_bus
+            assert global_act == ref_global_act
```
[the remark this answers is not in the document]
Comment (Farid-Najar) [reply grid2op/tests/test_ma.py]: done

File: grid2op/tests/test_ma.py [hunk 5]
```diff
@@ -597,6 +599,273 @@ def check_mask_topo_vect(self, ma_env, domain, space="action"):
...
+                global_storage_id = self.ma_env._subgrids_cls['action'][agent].storage_orig_ids[local_storage_id]
+                ref_global_act = self.ma_env._cent_env.action_space({
+                    'set_bus' : 
+                        (self.ma_env._cent_env.action_space.storage_pos_topo_vect[global_storage_id], 2)
+                })
+                assert (global_act.set_bus == ref_global_act.set_bus).all()
+                assert global_act._modif_set_bus == ref_global_act._modif_set_bus
+                assert global_act == ref_global_act
+
+    def test_local_action_to_global_change_bus(self):
```
[the remark this answers is not in the document]
Comment (Farid-Najar) [reply grid2op/tests/test_ma.py:729]: done

**Comment by BDonnot** [comment]
One feature at a time!

**Pull request closed without merging**